pkg/recovery/recover.go

```diff
@@ -0,0 +1,370 @@
+// Package recovery provides tooling to help with control plane disaster recovery. Recover() uses a
```

Comment:
Added a bunch of tests. It's not perfect but we can keep adding/updating as we improve the tool.